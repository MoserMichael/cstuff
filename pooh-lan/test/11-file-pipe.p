
f = openpipe( ~cmdline 'tr a A' )

f.write( ~towrite 'aaaaaaa' ~numwrite 7 )

f.closewrite()

output = emptystring()

f.read( ~toread output ~numread 7 )

status = f.wait()

println( ~msg 'output: [ output ] status: [ status ]' )




