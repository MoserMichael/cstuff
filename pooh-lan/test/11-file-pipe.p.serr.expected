002|... = openpipe( ~cmdline 'tr a A' )...
002|f = openpipe( ~cmdline 'tr a A' ):->(0x9585978)
004|f{'write'}:sub write(( ~towrite 'aaaaaaa' ~numwrite 7 )...
004|f{'write'}:sub write(( ~towrite 'aaaaaaa' ~numwrite 7 ):7
006|f{'closewrite'}:sub closewrite((  )...
006|f{'closewrite'}:sub closewrite((  ):Nill
008|... = emptystring(  )...
008|output = emptystring(  ):''
010|f{'read'}:sub read(( ~toread output:'' ~numread 7 )...
010|f{'read'}:sub read(( ~toread output:'' ~numread 7 ):7
012|... = f{'wait'}:sub wait((  )...
012|status = f{'wait'}:sub wait((  ):0
014|println( ~msg 'output: ' .. output:'AAAAAAA' .. ' status: ' .. status:0 .. '' )...
